src/styles/stars.scss

```diff
@@ -0,0 +1,24 @@
+.stars {
+  display: flex;
+
+  &__star {
+    width: 16px;
+    height: 16px;
+    margin-right: 4px;
+
+    background-image: url(../images/star.svg);
+    background-repeat: no-repeat;
+    background-position: center;
+  }
+
+  &__star:last-child {
+    margin-right: 0;
+  }
+  &.stars--1 .stars__star:nth-of-type(-n + 1),
+  &.stars--2 .stars__star:nth-of-type(-n + 2),
+  &.stars--3 .stars__star:nth-of-type(-n + 3),
+  &.stars--4 .stars__star:nth-of-type(-n + 4),
+  &.stars--5 .stars__star:nth-of-type(-n + 5) {
```

Comment:
You can optimize the SASS code by combining the selectors for the active stars. Instead of repeating the style for each modifier class, you can use a loop to generate the classes. This will make the code more maintainable and scalable. For example, you can use `@for $i from 1 through 5` to generate `.stars--#{$i} .stars__star:nth-of-type(-n + #{$i})`.